pkg/sql/gcjob_test/gc_job_test.go

```diff
@@ -317,6 +317,7 @@ SELECT job_id
   FROM [SHOW JOBS]
  WHERE job_type = 'SCHEMA CHANGE GC' AND description LIKE '%foo%';`,
 	).Scan(&jobID)
+	tdb.SucceedsSoonDuration = 3 * time.Minute
```

Comment:
Do we know why the GC job is taking so long? Increasing the timeout isn't addressing the root cause.

Reply:
it's not that the job is taking so long, it's just that it may take a while for the jobs framework to retry the job. the docs for retriable job errors say:

```
	// - retriable errors (see MarkAsRetryJobError) cause the job execution to be
	//   retried; Resume() will eventually be called again (perhaps on a different
	//   node).
```

i'm not sure what knobs there are (if any) to make that faster

Reply:
We have:
```
retryInitialDelaySettingKey    = "jobs.registry.retry.initial_delay"
retryMaxDelaySettingKey        = "jobs.registry.retry.max_delay"
```

Reply:
ah i see.. this test actually calls `params.Knobs.JobsTestingKnobs = jobs.NewTestingKnobsWithShortIntervals()` already, which should be setting those.

Reply:
Updated the PR to add more logging instead of increasing the timeout duration. Now we'll get some logging output like this:
```
    gc_job_test.go:322: Details about SCHEMA CHANGE GC job
    gc_job_test.go:323: job_id: 901466191613329409
    gc_job_test.go:324: status: running
    gc_job_test.go:325: last_run: 2023-09-20T02:14:09.311606Z
    gc_job_test.go:326: next_run: 2023-09-20T02:14:09.321606Z
    gc_job_test.go:327: num_runs: 1
```